### PR TITLE
feat: update Node.js in Docker to 20

### LINKED ISSUE #11290: [Feature]: Node 20 in puppeteer docker image
### Feature description

I would like to use Node 20 in the puppeteer image.
Node 20 is already in LTS status, and has some useful new shiny features.
I see no breaking changes in Node 20.

### PR DESCRIPTION
**Pull request by jrandolf-google**
Closes: https://github.com/puppeteer/puppeteer/pull/11289
Fixed: https://github.com/puppeteer/puppeteer/issues/11290

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:18@sha256:cd7fa8f136023f7500490e410ba70dd3982ccca21805264f2a260a3a97be7376
+FROM node:20@sha256:cd7fa8f136023f7500490e410ba70dd3982ccca21805264f2a260a3a97be7376
```
Comment (Lightning00Blade) [inline docker/Dockerfile]: SHA is not update.

Comment (jrandolf) [reply docker/Dockerfile]: Waiting for the run to finish.

**Review by Lightning00Blade — APPROVED** [review]

**Comment by ceck95** [comment]
i have question, hope can hear answer from you
why format image like "node:20@sha256:cb7cd40ba6483f37f791e1aace576df449fc5f75332c19ff59e2c6064797160e" ?
we can use format node:20.9.0 or some thing like that

**Comment by ceck95** [comment]
@jrandolf can you help me check on this

**Comment by Lightning00Blade** [comment]
@ceck95 You can read more about it here https://docs.docker.com/reference/cli/docker/image/pull/#pull-an-image-by-digest-immutable-identifier . But in short this insures users used the exact same version each time, make things easy to reproduce issues.

**Comment by ceck95** [comment]
yes thanks @Lightning00Blade , we can upgrade to node 20.13.1 (this is lts version), i tried by myself and this couldn't build

**Comment by Lightning00Blade** [comment]
@ceck95 If there is issue with the current Docker image please open a new issue with a description of what is happening and how to reproduce.

**Comment by ceck95** [comment]
okie @Lightning00Blade  appreciate for your answer my question

**Pull request merged**